docs/internal/release-process.md

```diff
@@ -0,0 +1,64 @@
+# Release Process Guide
+
+This guide is meant to explain the flow and rules of the release process, ie the period when we cut a release branch off the master branch in prevision of releasing it to users. Release Candidates (RC) are provided to test the Release.
+
+1. A release branch **must** be created at the start of the Release Process for the [Status App](https://github.com/status-im/status-app)'s master branch and [status-go](https://github.com/status-im/status-go)'s develop branch
+   1. The format for the status-app branch is `release/MAJOR.MINOR.x`, eg. `release/2.36.x`
+   2. The format for the status-go branch is automatically done using the Release script. Ask a status-go maintainer to create it.
+   3. The status-app release branch **must** always point to the status-go release branch.
+   
+2. Issues needing to be fixed on the RC **must** be added to the [Release Board](https://github.com/orgs/status-im/projects/97/views/1).
+   
+3. Only bug fixes **must** be provided to the release branch.
+   
+4. Moreover, only **critical** bug fixes **should** be added to the [Release Board](https://github.com/orgs/status-im/projects/97/views/1) and committed to the release branch.
+   1. Critical bugs are issues that affect the **behaviour** of the features.
+   2. This includes regressions to previously working features (eg: removing an emoji reaction no longer works).
+   3. This also includes issues breaking new features, ie **not** according to the expected Acceptance Criteria (eg: Browser is expected to allow saving bookmarks and the feature does not work).
+
+5. One or more Testing days **should** be done by the entire Status team to find any regressions.
+   1. The Status team **should** split in testing groups to help test features.
+   2. Testing groups **should** contain people from different teams to spread the knowledge.
+   3. Testing days **must** use a list of current features to test.
+   4. The list **should** include a column listing issues found and by whom.
+   5. The list **should** include an indicator whether that feature is covered by end-to-end (e2e) test and/or functional tests.
+   6. The list **should** contain a section documenting which features were recently **added** or **refactored**.
+   7. The recently added or refactored features **should** be the most tested.
+   8. Features on the list **can** be tested by multiple testing groups.
+
+6. Regressions **should** be mentioned to the QA team so that they can plan and implement e2e tests to prevent further regressions of the sorts (to be implemented on master).
+
+7. Code coverage does **not** need to be met on the release branch.
+
+8.  Features **must not** be allowed to be added to the release branch under any circumstances.
+
+9.  Fixes for issues identified during the RC phase **must** be worked on and committed on the release branch **first**.
+
+10. Once a fix has been applied to the release branch, it **must** be ported to the master branch.
```

Comment:
I would also clarify who is responsible for this step. Should the same developer who merged the commits into the release branch create the corresponding _cherry-pick PR_ to _master_ once it’s integrated? Or will there be a designated person during this phase to review everything added to the release and ensure it is properly ported back to _master_?